presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -381,6 +383,39 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     }
 
     @Override
+    public void createSchema(ConnectorSession session, String schemaName, Map<String, Object> properties)
+    {
+        Database database = new Database();
+        database.setName(schemaName);
+
+        HiveSchemaProperties.getLocation(properties).ifPresent(locationUri -> {
+            try {
+                hdfsEnvironment.getFileSystem(session.getUser(), new Path(locationUri));
+            }
+            catch (IOException e) {
+                throw new PrestoException(INVALID_SCHEMA_PROPERTY, "Invalid location URI: " + locationUri, e);
+            }
+            database.setLocationUri(locationUri);
+        });
+
+        PrincipalPrivilegeSet principalPrivilegeSet = buildInitialPrivilegeSet(session.getUser());
```

Comment:
I'm not sure what the right answer is.  In our code (https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java#L541) we set the permission based solely on the owner field, but the Thrift `Database` object has a `PrincipalPrivilegeSet` also, so I'm not sure what we should be doing.  Hive does support multiple pluggable authorization systems (e.g., https://cwiki.apache.org/confluence/display/Hive/Hive+Default+Authorization+-+Legacy+Mode), so maybe we should set both.